tqdm/_tqdm.py

```diff
@@ -1049,8 +1049,7 @@ def clear(self, nomove=False):
         if not nomove:
             self.moveto(self.pos)
         # clear up the bar (can't rely on sp(''))
-        self.fp.write('\r')
-        self.fp.write(' ' * (self.ncols if self.ncols else 10))
+        self.sp('')
```

Comment:
Oh, I totally missed the comment! All the unit tests pass fine and my own test scripts too, so I don't know nor remember at all why sp couldn't be relied on. I'm going to use git blame and git pickaxe to trace back this comment and see whether I can get more info.